### ISSUE TITLE
Enforcement of entity ordering

### ISSUE THREAD

**Issue by Lestropie** [bids-standard/legacy-validator#1390]
Relates to nipreps/fmriprep#2634.

I tested validator version `1.8.5` on a dataset that included files:

`sub-###/func/sub-###_task-rest_echo-#_run-#_bold.nii.gz`

According to the [MRI entity table](https://bids-specification.readthedocs.io/en/latest/99-appendices/04-entity-table.html#magnetic-resonance-imaging), `_run-#` should appear before `_echo-#`. In this circumstance, the `_run-#` entity was appended automatically by DICOM to BIDS conversion software. However no warning or error was raised by the validator.

Not very familiar with the validator so not aware of whether there is an existing test for entity ordering that is somehow not working, or if no such test is in place.

**Comment by rwblair** [comment]
The regex used for fMRI file name validation comes from this file:

https://github.com/bids-standard/bids-validator/blob/5cd727fdef7db52d20253f6a3efa5d19e58370b5/bids-validator/bids_validator/rules/file_level_rules.json#L249

I tried to reproduce the problem with dataset ds210 from the bids-standard/bids-example repository by renaming a single file to try and match the scheme mentioned:
`~/projects/bids-examples $ mv ./ds210/sub-06/func/sub-06_task-cuedSGT_run-03_echo-1_bold.nii.gz ./ds210/sub-06/func/sub-06_task-cuedSGT_echo-1_run-03_bold.nii.gz`

bvt is an alias to my local repository for the validator, when run it did complain:
```
~/projects/bids-examples $ bvt ds210
bids-validator@1.8.5
bids-specification@disable
        1: [ERR] Files with such naming scheme are not part of BIDS specification. This error is most commonly caused by typos in file names that make them not BIDS compatible. Please consult the specification and make sure your files are named correctly. If this is not a file naming issue (for example when including files not yet covered by the BIDS specification) you should include a ".bidsignore" file in your dataset (see https://github.com/bids-standard/bids-validator#bidsignore for details). Please note that derived (processed) data should be placed in /derivatives folder and source data (such as DICOMS or behavioural logs in proprietary formats) should be placed in the /sourcedata folder. (code: 1 - NOT_INCLUDED)
                ./sub-06/func/sub-06_task-cuedSGT_echo-1_run-03_bold.nii.gz
                        Evidence: sub-06_task-cuedSGT_echo-1_run-03_bold.nii.gz

```

I also tried running the validator again with the `-s v1.6.0` to generate the regex from the bids-specification schema files but it threw the same error.

Wonder what other differences there could be, anything standout that I missed? Just to double check what is the output of the validator and whats the command that you run?

**Comment by Lestropie** [comment]
Hmmm. I've not been able to reproduce myself either; even using the same version of the validator I used at the time, which turned out to be much more out of date than I thought. May have overlooked the generated errors due to being mixed in with other database issues. I'll close as a user error but keep an eye out for it.

(A dedicated error for erroneous entity ordering rather than a generic "not part of BIDS specification" would be beneficial; but that'd be a feature request rather than a bug report)

Sorry for the red herring.
